### PR TITLE
drivers: clock_control: Add API to set the clock rate

### PR DESCRIPTION
**Pull request by ceolin**
~Add a new API to set the clock rate synchronously and asynchronously.~

v2:
 - As discussed in the last PM meeting, the async API was removed and only synchronous API is implemented now.

### CODE REVIEW

**Comment by ceolin** [comment]
@teburd ^

**Review by teburd — APPROVED** [review]
I think this is fine, at some point we'll need a way to have things get notified on clock changes, but that's probably true for clock on/off as well, good stuff!

File: include/drivers/clock_control.h [hunk 1]
```diff
@@ -231,6 +236,43 @@ static inline int clock_control_get_rate(const struct device *dev,
...
+/**
+ * @brief Set the rate of the clock controlled by the device.
+ *
+ * On success, the new clock rate is set and ready when this function
+ * returns. This function may sleep, and thus can only be called from
+ * thread context.
+ *
+ * @param dev Device structure whose driver controls the clock.
+ * @param sys Opaque data representing the clock.
+ * @param rate Subsystem clock rate.
```
Comment (nordic-krch) [inline include/drivers/clock_control.h:248]: is this in Hz or opaque, vendor specific value?

Comment (ceolin) [reply include/drivers/clock_control.h:248]: I believe it is easier to use if it is vendor specific.

Comment (ceolin) [reply include/drivers/clock_control.h:248]: opinions ?

Comment (nordic-krch) [reply include/drivers/clock_control.h:248]: I also think that this could be vendor specific, So maybe rephrase to `Opaque data representing clock rate`

Comment (teburd) [reply include/drivers/clock_control.h:248]: Vendor specific makes sense to me as well

Comment (ceolin) [reply include/drivers/clock_control.h]: cool ! Thanks for the feedback @teburd.

@nordic-krch can you take a look on it again ?

**Review by nordic-krch — APPROVED** [review]
